inc/wp-components/classes/class-component.php

```diff
@@ -73,7 +73,12 @@ class Component implements \JsonSerializable {
 	 * Component constructor.
 	 */
 	public function __construct() {
-		$this->config   = $this->default_config();
+
+		// Set default configs using set_config to ensure callbacks fire.
+		foreach ( $this->default_config() as $key => $value ) {
```

Comment:
Is the actual callback for `theme_name_set_config` supposed to be included in this PR too?